src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -108,6 +110,7 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 				}
 			}
 		}));
+		this._register(lifecycleService.onWillShutdown(() => this._disposeAllDecorations()));
```

Comment:
I don't understand the fix here, was this problem only happening on reconnected terminal or something?

Reply:
yes

Reply:
https://github.com/microsoft/vscode/issues/155446#issuecomment-1203155690

Reply:
cool, thanks